lib/data/external/GCP/GcpService.js

```diff
@@ -1,16 +1,39 @@
 const googleAuth = require('google-auto-auth');
+const async = require('async');
 const AWS = require('aws-sdk');
 const { errors } = require('arsenal');
 const Service = AWS.Service;
 
 const GcpSigner = require('./GcpSigner');
 
-function genAuth(authParams) {
-    const authOptions = authParams || {};
-    Object.assign(authOptions, {
-        scopes: 'https://www.googleapis.com/auth/cloud-platform',
-    });
-    return googleAuth(authOptions);
+function genAuth(authParams, callback) {
+    async.waterfall([
+        function authKeyFile(next) {
+            const authOptions = {
+                scopes: authParams.scopes,
+                keyFilename: authParams.keyFilename,
+            };
+            const auth = googleAuth(authOptions);
+            auth.getAuthClient(err => {
```

Comment:
I see anti-patterns here. Why is the `err` being ignored? Also, please use error-first callbacks.

Reply:
Cleaned up the err issue with `async.tryEach`